pkg/transport/kafka/kafka_consumer_client.go

```diff
@@ -0,0 +1,292 @@
+package kafka
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"github.com/go-logr/logr"
+	datatypes "github.com/open-cluster-management/hub-of-hubs-data-types"
+	kafkaconsumer "github.com/open-cluster-management/hub-of-hubs-kafka-transport/kafka-client/kafka-consumer"
+	kafkaHeaderTypes "github.com/open-cluster-management/hub-of-hubs-kafka-transport/types"
+	compressor "github.com/open-cluster-management/hub-of-hubs-message-compression"
+	"github.com/open-cluster-management/hub-of-hubs-message-compression/compressors"
+	"github.com/open-cluster-management/leaf-hub-spec-sync/pkg/bundle"
+	"github.com/open-cluster-management/leaf-hub-spec-sync/pkg/transport"
+)
+
+const (
+	commitRescheduleDelay = time.Second * 20
+	bufferedChannelSize   = 42
+)
+
+var (
+	errReceivedUnsupportedBundleType = errors.New("received unsupported message type")
+	errMissingCompressionType        = errors.New("compression type is missing from message headers")
+)
+
+// NewConsumer creates a new instance of Consumer.
+func NewConsumer(log logr.Logger, bundleUpdatesChan chan *bundle.ObjectsBundle) (*Consumer, error) {
```

Comment:
see all comments in this file from previous PR.. just open the file and drill down to see all comments:
https://github.com/open-cluster-management/leaf-hub-spec-sync/pull/6/files